File: solutions/go/bob/1/bob.go
// Package bob is about bob the teenager, who communicates in cryptic messages.
package bob

import "strings"

// Hey is an example of an artificially intelligent simulator of a teenager's brain.
func Hey(remark string) string {
	trimmed := strings.TrimSpace(remark)

	if trimmed == "" {
		return "Fine. Be that way!"
	}

	isQuestion := trimmed[len(trimmed)-1] == '?'
	isYelling := strings.ToUpper(trimmed) == trimmed && strings.ToLower(trimmed) != trimmed

	if isQuestion && isYelling {
		return "Calm down, I know what I'm doing!"
	}

	if isQuestion {
		return "Sure."
	}

	if isYelling {
		return "Whoa, chill out!"
	}

	return "Whatever."
}

File: solutions/typescript/bob/1/bob.ts
export function hey(message: string): string {
  const trimmed = message.trim()
  
  const isQuestion = trimmed[trimmed.length - 1] == "?"
  const isShouting = trimmed == trimmed.toUpperCase() && trimmed != trimmed.toLowerCase()
  
  if ( isQuestion ){
    if( isShouting ){
      return "Calm down, I know what I'm doing!"
    }
    return "Sure."
  }
  if(trimmed == ""){
    return "Fine. Be that way!"
  }
  if(isShouting){
    return "Whoa, chill out!"
  }
  return "Whatever."
}
